src/main/java/ch/njol/skript/expressions/ExprDurability.java

```diff
@@ -89,62 +86,67 @@ public Class<?>[] acceptChange(ChangeMode mode) {
 
 	@Override
 	public void change(Event event, @Nullable Object[] delta, ChangeMode mode) {
-		int i = delta == null ? 0 : ((Number) delta[0]).intValue();
-		Object[] objects = getExpr().getArray(event);
-		for (Object object : objects) {
-			ItemStack itemStack = null;
-
-			if (object instanceof ItemType) {
-				itemStack = ((ItemType) object).getRandom();
-			} else if (object instanceof Slot) {
-				itemStack = ((Slot) object).getItem();
-			}
-			if (itemStack == null)
-				return;
-
-			int changeValue = ItemUtils.getDamage(itemStack);
-			if (durability)
-				changeValue = itemStack.getType().getMaxDurability() - changeValue;
-
+		int change = delta == null ? 0 : ((Number) delta[0]).intValue();
+		if (mode == ChangeMode.REMOVE)
+			change = -change;
+		for (Object object : getExpr().getArray(event)) {
+			ItemType itemType = asItemType(object);
+			if (itemType == null)
+				continue;
+
+			ItemMeta meta = itemType.getItemMeta();
+			if (!(meta instanceof Damageable))
+				continue;
+			Damageable damageable = (Damageable) meta;
+
+			Material material = itemType.getMaterial();
 			switch (mode) {
-				case REMOVE:
-					i = -i;
-					//$FALL-THROUGH$
 				case ADD:
-					changeValue += i;
+				case REMOVE:
+					int current = convertToDamage(material, damageable.getDamage());
+					damageable.setDamage(convertToDamage(material, current + change));
 					break;
 				case SET:
-					changeValue = i;
+					damageable.setDamage(convertToDamage(material, change));
 					break;
 				case DELETE:
 				case RESET:
-					changeValue = 0;
-					break;
-				case REMOVE_ALL:
-					assert false;
+					damageable.setDamage(0);
 			}
 
-			if (durability && mode != ChangeMode.RESET && mode != ChangeMode.DELETE)
-				changeValue = itemStack.getType().getMaxDurability() - changeValue;
-
-			if (object instanceof ItemType) {
-				ItemUtils.setDamage(itemStack, changeValue);
-				((ItemType) object).setTo(new ItemType(itemStack));
-			} else {
-				ItemUtils.setDamage(itemStack, changeValue);
-				((Slot) object).setItem(itemStack);
-			}
+			itemType.setItemMeta(meta);
+			if (object instanceof Slot)
+				((Slot) object).setItem(itemType.getRandom());
 		}
 	}
 
+	private int convertToDamage(Material material, int value) {
+		if (!durability)
+			return value;
+		int maxDurability = material.getMaxDurability();
+		if (maxDurability == 0)
+			return 0;
+		return maxDurability - value;
```

Comment:
```suggestion
		return Math.max(material.getMaxDurability() - value, 0);
```
Would be nice :) (unless negative values are intended?)

Reply:
I am not actually sure if a negative durability value is valid. If they are, then this does not need to be done.

Reply:
You can certainly have an item with negative durability, it just looks stupid in game
I tested it out when I reviewed this